### PR TITLE
[ENG-973] Consider TrustedForm cert URLs that do not end in an SHA1 as invalid.

### PR DESCRIPTION
**Pull request by heathdutton**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: Integration/TrustedFormIntegration.php [hunk 1]
```diff
@@ -65,6 +65,8 @@ public function doEnhancement(Lead $lead)
                 || 'https' !== $parts['scheme']
                 || !isset($parts['host'])
                 || self::CERT_REAL_HOST !== $parts['host']
+                || !isset($parts['path'])
+                || !preg_match('/^\/[0-9a-f]{40}$/i', $parts['path'])
```
Comment (ghost) [inline Integration/TrustedFormIntegration.php:69]: RegEx 🔥 💯

File: Integration/TrustedFormIntegration.php [hunk 2]
```diff
@@ -156,9 +163,12 @@ public function doEnhancement(Lead $lead)
                                 $lead->addUpdatedField('trusted_form_share_url', $data->share_url);
                                 $persist = true;
                             }
-                            $this->logger->info(
-                                'TrustedForm: Contact '.$identifier.' '.(!$persist ? 'NOT ' : '').'updated. '.(!empty($data->message) ? $data->message : '')
-                            );
+
+                            if ($persist) {
+                                $this->logger->info(
+                                    'TrustedForm: Contact '.$identifier.' '.(!$persist ? 'NOT ' : '').'updated. '.(!empty($data->message) ? $data->message : '')
```
Comment (ghost) [inline Integration/TrustedFormIntegration.php:169]: Ternary operator here is redundant, $persist is already true

File: Tests/Integration/TrustedFormIntegrationTest.php [hunk 3]
```diff
@@ -70,7 +70,7 @@ public function testDoEnhancement20XResponse()
         $this->leadObserver->expects($this->any())
             ->method('getFieldValue')
             ->willReturnMap([
-                ['xx_trusted_form_cert_url', null, 'https://cert.trustedform.com'],
+                ['xx_trusted_form_cert_url', null, 'https://cert.trustedform.com/a94a8fe5ccb19ba61c4c0873d391e987982fbbd3'],
```
Comment (dean-martin) [inline Tests/Integration/TrustedFormIntegrationTest.php]: Does this test rely on this hash to pass?

**Review by ghost — DISMISSED** [review]
Just a redundant ternary operator.

**Review by westonwatson — APPROVED** [review]

File: Integration/TrustedFormIntegration.php (hunk 2 above)
Comment (heathdutton) [reply Integration/TrustedFormIntegration.php:169]: True, let's fix next time we swing through this.

**Review by scottshipman — APPROVED** [review]

**Pull request merged**